site-builder/src/site/resource.rs

```diff
@@ -458,3 +487,118 @@ pub(crate) fn full_path_to_resource_path(full_path: &Path, root: &Path) -> Resul
+            },
+            PatternMatchTestCase {
+                pattern: "test/*",
+                path: "test/file",
+                expected: true,
+            },
+            PatternMatchTestCase {
+                pattern: "test/*",
+                path: "test/file.extension",
+                expected: true,
+            },
+            PatternMatchTestCase {
+                pattern: "test/*",
+                path: "test/foo.bar.extension",
+                expected: true,
+            },
+            PatternMatchTestCase {
+                pattern: "test/*",
+                path: "test/foo-bar_baz.extension",
+                expected: true,
+            },
+            PatternMatchTestCase {
+                pattern: "[invalid",
+                path: "file",
+                expected: false,
+            },
+        ];
+        for t in tests {
+            assert_eq!(
+                ResourceManager::is_pattern_match(t.pattern, t.path),
+                t.expected
+            );
+        }
+    }
+
+    #[ignore = "The test depends on the file system containing a walrus binary.
+Until we find a way to mock the walrus binary, this test will be ignored."]
+    #[tokio::test]
+    async fn test_derive_http_headers() {
+        let resource_manager = setup_resource_manager_mock().await;
+        let test_paths = vec![
+            // This is the longest path. So `/foo/bar/baz/*.svg` would persist over `*.svg`.
+            ("/foo/bar/baz/image.svg", "etag"),
+            // This will only match `*.svg`.
+            (
+                "/very_long_name_that_should_not_be_matched.svg",
+                "cache-control",
+            ),
+        ];
+        for (path, expected) in test_paths {
+            let result = resource_manager.derive_http_headers(path);
+            assert_eq!(result.len(), 1);
+            assert!(result.contains_key(expected));
+        }
+    }
+
+    /// Sets up a mock resource manager with the given headers configuration and resource path.
+    ///
+    /// Helper function for testing the `derive_http_headers` method.
+    async fn setup_resource_manager_mock() -> ResourceManager {
+        let walrus_mock = Walrus::new("walrus".to_string(), 1234, None, None, None);
+        // Define the headers configuration for mocking the resource manager.
+        let mut headers: BTreeMap<String, HttpHeaders> = BTreeMap::new();
+        headers.insert(
+            "/*.svg".to_string(),
+            HttpHeaders(BTreeMap::from([(
+                "cache-control".to_string(),
+                "public, max-age=86400".to_string(),
+            )])),
+        );
+        headers.insert(
+            "/foo/bar/baz/*.svg".to_string(),
+            HttpHeaders(BTreeMap::from([(
+                "etag".to_string(),
+                "\"abc123\"".to_string(),
+            )])),
+        );
+        let ws_resources = Some(WSResources {
+            headers: Some(headers),
+            routes: None,
+        });
```

Comment:
Suggestion: may be cleaner to provide a json an deserialize from it (see tests in `config.rs`)

Reply:
https://github.com/MystenLabs/walrus-sites/pull/272/commits/88be590bb0868b6785254444ec50cddc1ef1f701